include/cetl/pmr/memory.hpp

```diff
@@ -0,0 +1,231 @@
+/// @file
+/// Extensions and utilities for types found in the standard `memory` header to better integrate with pmr types.
+///
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+///
+
+#ifndef CETL_PMR_MEMORY_H_INCLUDED
+#define CETL_PMR_MEMORY_H_INCLUDED
+
+#ifndef CETL_H_ERASE
+#    include "cetl/cetl.hpp"
+#endif
+
+#include <memory>
+#include <utility>
+
+namespace cetl
+{
+namespace pmr
+{
+
+/// RAII helper for memory_resource::allocate() and memory_resource::deallocate(). This type is designed to work with
+/// std::unique_ptr.
+///
+/// Example usage:
+///
+/// @snippet{trimleft} example_06_memory_resource_deleter.cpp example_usage
+///
+class MemoryResourceDeleter final
```

Comment:
Does a full class definition in the header violate the One Definition Rule?  Do you have to put "inline" or the methods for it to be correct in a header-only library?

Reply:
>Does a full class definition in the header violate the One Definition Rule?

No.

>Do you have to put "inline" or the methods for it to be correct in a header-only library?

Member functions defined inside the class body are always implicitly declared `inline`.